recipe/meta.yaml

```diff
@@ -27,42 +27,131 @@ source:
+    test:
+      requires:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - cmake
+        - ninja
+      files:
+        - test.cu
+        - CMakeLists.txt
+      commands:
+        - nvcc --version
+        - nvcc --verbose test.cu
+        - cmake -S . -B ./build -G=Ninja && cmake --build ./build -v
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: Compiler for CUDA applications.
+      description: |
+        Compiler for CUDA applications.
+      doc_url: https://docs.nvidia.com/cuda/index.html
 
-test:
-  requires:
-    - {{ compiler("c") }}
-    - {{ compiler("cxx") }}
-    - cmake >=3.20
-    - ninja
-  files:
-    - test.cu
-    - CMakeLists.txt
-  commands:
-    - test -f $PREFIX/targets/{{ target_name }}/lib/libnvptxcompiler_static.a  # [linux]
-    - test -f $PREFIX/targets/{{ target_name }}/include/nvPTXCompiler.h        # [linux]
-    - if not exist %LIBRARY_LIB%\nvptxcompiler_static.lib exit 1  # [win]
-    - if not exist %LIBRARY_INC%\nvPTXCompiler.h exit 1           # [win]
-    - nvcc --version
-    - nvcc --verbose test.cu
-    - cmake -S . -B ./build -G=Ninja && cmake --build ./build -v
+  - name: cuda-nvcc_{{ target_platform }}
+    files:
+      - etc/conda/activate.d/~cuda-nvcc_activate.sh      # [linux]
+      - etc/conda/deactivate.d/~cuda-nvcc_deactivate.sh  # [linux]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+        - cuda-cudart-dev_{{ target_platform }}
+      run:
+        - cuda-cudart-dev_{{ target_platform }}
+        - {{ pin_subpackage("cuda-nvcc-impl_" + target_platform, exact=True) }}
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}    # [aarch64]
+        - cmake >=3.20
+    test:
+      commands:
+        - test -f $PREFIX/etc/conda/activate.d/~cuda-nvcc_activate.sh      # [linux]
+        - test -f $PREFIX/etc/conda/deactivate.d/~cuda-nvcc_deactivate.sh  # [linux]
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: Compiler for CUDA applications.
+      description: |
+        Compiler for CUDA applications.
+      doc_url: https://docs.nvidia.com/cuda/index.html
+
+  - name: cuda-nvcc-impl_{{ target_platform }}
+    build:
+      noarch: generic
+      run_exports:
+        strong:
+          - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
```

Comment:
```suggestion
  - name: cuda-nvcc-impl_{{ target_platform }}
    build:
      noarch: generic
      missing_dso_whitelist:
        # Need to avoid missing linkage errors due to `noarch`.
        # Library dependencies are pulled in.
        - "*"
    run_exports:
      strong:
        - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
```